packages/runtime/src/components/builtin/Box/Box.tsx

```diff
@@ -123,22 +129,61 @@ const Box = forwardRef(function Box(
     responsiveStyle([verticalAlign], ([alignItems = 'flex-start']) => ({ alignItems })),
   )
 
-  const { initial, animate, variants, transition, key } = useBoxAnimations({
-    boxAnimateType,
-    boxAnimateDuration,
-    boxAnimateDelay,
-    itemAnimateDelay,
-    itemAnimateType,
-    itemAnimateDuration,
-    itemStaggerDuration,
-    boxElement,
-    elements: children?.elements,
-  })
+  const key = useMemo(
+    () => v4(),
+    [
+      boxAnimateType,
+      boxAnimateDuration,
+      boxAnimateDelay,
+      itemAnimateType,
+      itemAnimateDuration,
+      itemAnimateDelay,
+      itemStaggerDuration,
+    ],
+  )
```

Comment:
This seems like a very weird way to use `useMemo` since none of the dependencies are used in computing the memoized value. Did you explore other ways to get a key?

Reply:
Yeah I did. I don't know of a better way of watching props as a user changes them. and rerendering the children.

Reply:
How about being explicit about the different in the props and calling `setState` directly in render? This is fine and is something I've seen used in the React docs

Reply:
I think I am forgoing this solution. In my testing the `setState` in render doesn't cause the `useEffect(() => {}, [key])` to rerun.

Reply:
That sounds very weird to me and possibly a fluke where something else could be causing that (i.e., the `key` not changing). But if you feel like `useMemo` is the way to go, I'm not going to push for something else

Reply:
Actually I could just put. this into the render too 

```
  useEffect(() => {
    if (hasIntersected.current === false) return

    boxElement?.classList.remove('show')
    const timeout = setTimeout(() => boxElement?.classList.add('show'), 100)

    return () => clearTimeout(timeout)
  }, [key])
  ```

Reply:
Actually that doesn't work either. I think this is why doing stuff in a ref callback is an anti pattern :/

Reply:
Why would it be an anti-pattern?